### PR TITLE
feat: Enable Organizer to resend invitation

### LINKED ISSUE #5416: Team Page: Enable Organizer to resend invitation
* On the team page of an event please enable the organizer to resend an invitation. 
* Add a button "Resend invitation" if the user has not accepted the invitation yet.

Screenshot from 2020-10-29 11-46-19

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #5416 

 require https://github.com/fossasia/open-event-server/pull/7522

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, review basic structure. Error handling is remaining

**Comment by iamareebjamal** [comment]
Should be a POST request

**Comment by sachinchauhan2889** [comment]
> Should be a POST request

Ok sir i will do it

File: app/components/events/view/overview/manage-roles.js [hunk 1]
```diff
@@ -54,7 +54,16 @@ export default class ManageRoles extends Component {
 
   @action
   async resendInvite(invite) {
-    await this.loader.load('/role-invites/'+invite.id+'/resend-invite');
+    try {
+      await this.loader.post('/role-invites/' + invite.id + '/resend-invite');
```
Comment (iamareebjamal) [inline app/components/events/view/overview/manage-roles.js:58]: Check the response and display success if it was successful

File: app/components/events/view/overview/manage-roles.js [hunk 2]
```diff
@@ -54,7 +54,16 @@ export default class ManageRoles extends Component {
...
+      this.notify.success(this.l10n.t('Invite resent successfully'),
+        {
+          id: 'resend_invite_succ'
+        });
+    } catch (error) {
+      console.log(error);
```
Comment (iamareebjamal) [inline app/components/events/view/overview/manage-roles.js:64]: Check how other errors are handled

File: app/components/events/view/overview/manage-roles.js [hunk 3]
```diff
@@ -53,17 +53,31 @@ export default class ManageRoles extends Component {
...
-      await this.loader.post('/role-invites/' + invite.id + '/resend-invite');
-      this.notify.success(this.l10n.t('Invite resent successfully'),
-        {
-          id: 'resend_invite_succ'
-        });
-    } catch (error) {
-      console.log(error);
-      this.notify.error(this.l10n.t('Oops something went wrong. Please try again'));
-    }
+  resendInvite(invite) {
```
Comment (iamareebjamal) [inline app/components/events/view/overview/manage-roles.js:56]: Make it async

File: app/components/events/view/overview/manage-roles.js [hunk 4]
```diff
@@ -53,17 +53,31 @@ export default class ManageRoles extends Component {
...
+    this.set('isLoading', true);
+    this.loader
```
Comment (iamareebjamal) [inline app/components/events/view/overview/manage-roles.js:58]: await

File: app/components/events/view/overview/manage-roles.js [hunk 5]
```diff
@@ -53,17 +53,31 @@ export default class ManageRoles extends Component {
...
+      .post('/role-invites/' + invite.id + '/resend-invite')
+      .then(res => {
+        if (res.success) {
```
Comment (iamareebjamal) [inline app/components/events/view/overview/manage-roles.js:61]: Why going backwards? Why switching from async await to promises?

File: app/components/events/view/overview/manage-roles.js [hunk 6]
```diff
@@ -53,17 +53,31 @@ export default class ManageRoles extends Component {
...
+              id: 'resend_invite_succ'
+            });
+        } else {
+          this.notify.error(this.l10n.t('Oops something went wrong. Please try again'));
+        }
+      })
+      .catch(error => {
+        console.error('Error while resending invite', error);
+        if (error.status === 403) {
+          this.notify.error(this.l10n.t('You are not allowed to resend invite. Only organizers can resend invite.'));
```
Comment (iamareebjamal) [inline app/components/events/view/overview/manage-roles.js:73]: This page is only visible to organizers, just show oops error

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**